source/common/config/grpc_mux_impl.h

```diff
@@ -122,6 +123,9 @@ class NullGrpcMuxImpl : public GrpcMux {
   }
   void pause(const std::string&) override {}
   void resume(const std::string&) override {}
+  bool paused(const std::string&) const override {
+    throw EnvoyException("ADS must be configured to support an ADS config source");
```

Comment:
If this is a valid config mistake, can you add a test for this exception? If this can't happen can you just NOT_REACHED this?

Reply:
Ah right yes, much more sensible; done.